api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/processor/CompletePaymentMessageProcessorTest.java

```diff
@@ -0,0 +1,89 @@
+    @Mock
+    private PaymentService paymentService;
+    @Mock
+    private PaymentCycleNotificationHandler paymentCycleNotificationHandler;
+
+    @InjectMocks
+    private CompletePaymentMessageProcessor processor;
+
+    @Test
+    void shouldProcessMessageAndSendRegularPaymentEmail() {
+        PaymentCycle paymentCycle = aValidPaymentCycle();
+        Claim claim = paymentCycle.getClaim();
+        CompletePaymentMessageContext messageContext = aValidCompletePaymentMessageContextForScheduledPayment(paymentCycle, claim);
+        given(messageContextLoader.loadCompletePaymentMessageContext(any())).willReturn(messageContext);
+        Message message = aValidMessageWithType(COMPLETE_PAYMENT);
+
+        MessageStatus result = processor.processMessage(message);
+
+        assertThat(result).isEqualTo(MessageStatus.COMPLETED);
+        verify(messageContextLoader).loadCompletePaymentMessageContext(message);
+        verify(paymentService).completePayment(paymentCycle, messageContext.getPaymentCalculation(), messageContext.getPaymentResult());
+        verify(paymentCycleNotificationHandler).sendNotificationEmailsForRegularPayment(paymentCycle);
+    }
+
+    @Test
+    void shouldProcessMessageAndSendRestartedPaymentEmail() {
+        PaymentCycle paymentCycle = aValidPaymentCycle();
+        Claim claim = paymentCycle.getClaim();
+        CompletePaymentMessageContext messageContext = aValidCompletePaymentMessageContextForRestartedPayment(paymentCycle, claim);
+        given(messageContextLoader.loadCompletePaymentMessageContext(any())).willReturn(messageContext);
+        Message message = aValidMessageWithType(COMPLETE_PAYMENT);
+
+        MessageStatus result = processor.processMessage(message);
+
+        assertThat(result).isEqualTo(MessageStatus.COMPLETED);
+        verify(messageContextLoader).loadCompletePaymentMessageContext(message);
+        verify(paymentService).completePayment(paymentCycle, messageContext.getPaymentCalculation(), messageContext.getPaymentResult());
+        verify(paymentCycleNotificationHandler).sendNotificationEmailsForRestartedPayment(paymentCycle);
+    }
+
+    @Test
+    void shouldProcessMessageAndSendNoEmailsForInitialPayment() {
+        PaymentCycle paymentCycle = aValidPaymentCycle();
+        Claim claim = paymentCycle.getClaim();
+        CompletePaymentMessageContext messageContext = aValidCompletePaymentMessageContextForFirstPayment(paymentCycle, claim);
+        given(messageContextLoader.loadCompletePaymentMessageContext(any())).willReturn(messageContext);
+        Message message = aValidMessageWithType(COMPLETE_PAYMENT);
+
+        MessageStatus result = processor.processMessage(message);
+
+        assertThat(result).isEqualTo(MessageStatus.COMPLETED);
+        verify(messageContextLoader).loadCompletePaymentMessageContext(message);
+        verify(paymentService).completePayment(paymentCycle, messageContext.getPaymentCalculation(), messageContext.getPaymentResult());
```

Comment:
These 3 lines are duplicated in all tests in this class, is it worthwhile refactoring into a common method?

Reply:
Personally I'm ok with this level of duplication in tests. If the duplicate code grows then I might consider.